services:
  db:
    container_name: 'pizza_db'
    image: postgres:16.4
    restart: always
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - "5437:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - app_network
    healthcheck:
      test: [ "CMD", "pg_isready", "-U", "${POSTGRES_USER}", "-d", "${POSTGRES_DB}" ]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    container_name: 'pizza_app'
    build:
      context: .
      dockerfile: Dockerfile
    depends_on:
      db:
        condition: service_healthy
    env_file:
      - .env
    ports:
      - "8081:8000"
    volumes:
      - .:/pizza_app
    networks:
      - app_network
    restart: always
    command: >
      bash -c "
      python rest_pizza_delivery/db/db_init.py &&
      uvicorn rest_pizza_delivery.main:app --host 0.0.0.0 --port 8000"

volumes:
  postgres_data:

networks:
  app_network:
    driver: bridge

